File: frontend/src/api/chat/dto/GetChatRoom.ts
export interface GetChatRoomResponse {
  id: number
  rentalItem: RentalItem
  participants: Participant[]
}

interface RentalItem {
  id: number
  seller: Seller
  category: string
  title: string
  pricePerDay?: number
  pricePerWeek?: number
  thumbnailImageUrl: string
}

interface Seller {
  id: number
  nickname: string
  profileImageUrl?: string
  address: string
}

export interface Participant {
  id: number
  nickname: string
  profileImageUrl?: string
}

File: backend/api/src/main/kotlin/io/github/daegwonkim/backend/dto/chatroom/GetChatRoom.kt
package io.github.daegwonkim.backend.dto.chatroom

import io.github.daegwonkim.backend.enumerate.RentalItemCategory

data class GetChatRoomResponse(
    val id: Long,
    val rentalItem: RentalItem,
    val participants: List<Participant>
) {
    data class RentalItem(
        val id: Long,
        val seller: Seller,
        val category: RentalItemCategory,
        val title: String,
        val pricePerDay: Int?,
        val pricePerWeek: Int?,
        val thumbnailImageUrl: String
    ) {
        data class Seller(
            val id: Long,
            val nickname: String,
            val profileImageUrl: String?,
            val address: String
        )
    }

    data class Participant(
        val id: Long,
        val nickname: String,
        val profileImageUrl: String?
    )
}